tests/frameworks/OSXShims/src/LoggingNative.mm

```diff
@@ -15,44 +15,92 @@
 //******************************************************************************
 
 #import <LoggingNative.h>
+#import <Starboard/SmartTypes.h>
 #import <Foundation/NSString.h>
 
 #import <cstdarg>
 
+// NOTE: this function is assumed to only run in Little Endian environments like OSX
+static NSStringEncoding _getWcharEncoding() {
+    switch (sizeof(wchar_t)) {
+        case 2:
+            return NSUTF16LittleEndianStringEncoding;
+        case 4:
+            return NSUTF32LittleEndianStringEncoding;
+        default:
+            assert(false);
+    }
+}
+
+static NSStringEncoding getWcharEncoding() {
+    static NSStringEncoding encoding = _getWcharEncoding();
+    return encoding;
+}
+
 void TraceVerbose(const wchar_t* tag, const wchar_t* format, ...) {
-    NSString* formatString = [NSString stringWithFormat:@"[%ls] %ls", tag, format];
+    StrongId<NSString> nsTag;
```

Comment:
justification? it would be half as long.

---
In reply to: [74170322](https://github.com/Microsoft/WinObjC/pull/740#discussion_r74170322) [](ancestors = 74170322,74169017)